api/config/v1alpha1/envoygateway_types.go

```diff
@@ -96,7 +96,7 @@ type Extension struct {
 	Name string
 
 	// APIGroups defines the set of K8s api groups the extension will handle.
-	APIGroups []string
+	APIGroups []APIGroup
```

Comment:
https://pkg.go.dev/sigs.k8s.io/gateway-api@v0.6.0/apis/v1beta1#Group

Same do we just leverage the Gateway-API

Reply:
Well `EnvoyGateway` isn't a real CRD but I think that's something worth asking to the rest of the Envoy Gateway team.

For V0, I think it will be easier to create our own types as needed.